test/lightning/apollo_client_test.exs

```diff
@@ -49,4 +50,20 @@ defmodule Lightning.ApolloClientTest do
 
     assert response.body
   end
+
+  test "send a query and get a timeout acording to apollo config" do
+    Mox.stub(Lightning.MockConfig, :apollo, fn key ->
+      case key do
+        :endpoint -> "http://localhost:3000"
+        :ai_assistant_api_key -> "api_key"
+        :timeout -> 100
+      end
+    end)
+
+    stub(Lightning.Tesla.Mock, :call, fn _env, _opts ->
+      Process.sleep(200)
```

Comment:
Is it okay to have process sleep operations in our test suite ? How would this impact the speed of our tests ? I know that tests are ran async but just wanna make sure

Reply:
A very short one is acceptable to cause the situation where the call takes more time than the timeout set. The call takes 200ms for a timeout config of 100ms so that in this test case in particular gonna return the timeout error.

Reply:
It was needed because the :tesla app :adapter for test is the `Lightning.Tesla.Mock`. But now the timeout takes effect only on Finch so I have removed the test case.

I could put a different adapter only for this test case but we would risk introducing a flaky as the temporary config, with Finch adapter during the test, could be read by another test file.